unyt/_array_functions.py

```diff
@@ -186,3 +186,83 @@ def histogramdd(
         density=density,
     )
     return counts, tuple(_bin * u for _bin, u in zip(bins, units))
+
+
+def _validate_units_consistency(arrs):
+    # NOTE: we cannot validate that all arrays are unyt_arrays
+    # by using this as a guard clause in unyt_array.__array_function__
+    # because it's already a necessary condition for numpy to use our
+    # custom implementations
+    a1 = arrs[0]
+    if not all(a.units == a1.units for a in arrs[1:]):
+        raise RuntimeError("Your arrays must have identical units.")
```

Comment:
Should this be `UnitOperationError` or one of the other unyt-specific errors?

Reply:
You tell me. Here I basically copied and adapted the content of `unyt.array._validate_numpy_wrapper_units`. I'm happy to change this to any custom error you deem more appropriate.

Reply:
I think both should probably be `UnitOperationError`, if the other one is a `RuntimeError` that's an oversight. Ideally users shouldn't need to guess which errors `unyt` will generate for a given operation, and `UnitOperationError` is what we raise elsewhere when a user tries to do an operation with invalid units.

Reply:
Ah, I was a little careless when I did this and didn't realise that `UnitOperationError.__init__` had a template error message. Because it refers to an "operator", I'm not sure the wording works that well in the context of array functions. Also, `UnitOperationError` takes 2 units at most, but here we might have an arbitrary number of them. Should I subclass it to a more finely tailored `UnitInconstencyError` class ?

Reply:
Sure, that make sense. Sorry, didn't realize that wouldn't just be a straightforward substitution.

Reply:
no worries